controller/controller.go

```diff
@@ -237,7 +238,7 @@ func (c *Controller) ProcessEvents(ctx context.Context, deleteChan chan watch.Ev
 				c.UpdateFunc(nil, e.Object)
 				t.ObserveDuration()
 			case err := <-errChan:
-				return microerror.Mask(err)
+				return microerror.Mask(matchAndTransform(err, c.crd.Name))
```

Comment:
So why is that different now to the implementation you had before in the errors lib? I liked that a lot. There was the extended matcher with tests and here we introduce a completely new way of error handling. I don't see why we should to that. IMO this would be the better and already established way. 

```go
if IsStatusForbidden(err) {
  return microerror.Maskf(statusForbiddenError, "controller might be missing RBAC rule for %s CRD", c.crd.Name)
} else if err != nil {
  return microerror.Mask(err)
}
```

Reply:
Because that piece of code doesn't differentiate between k8s error and our error but groups them together hiding away which error we really matched. Matching against an error `x` and then returning a new instance of it looks weird. There are two reasons why I extracted it into separate function:
- Keep this piece of code concise as it was
- Provide foundation for future additions with minimal changes. This is directly aligned with our earlier approach to e.g. divide empty `[]Resource{}` to multiple lines instead of keeping it as that. If there are more matchers added here, this section gets very quickly cluttered and ugly.

Reply:
> Matching against an error `x` and then returning a new instance of it looks weird.

This is what you already implemented. The code is now just in line 241 where it belongs to IMO and not in line 454 like in your suggestion. What you suggested is not how we do error handling usually and I am against introducing a new way of doing things. I do not see the benefit of changing the way we do it already. 

> f there are more matchers added here, this section gets very quickly cluttered and ugly.

With your suggestion it will in `matchAndTransform`. With my suggestion it will in `ProcessEvents`. The only difference I see is the inconsistency of the error handling approach you introduce now and I am against that because I do not see a reason why we should do this.

Reply:
PTAL